src/pages/permissions/panels/EditGroupPermissionsPanel.tsx

```diff
@@ -0,0 +1,481 @@
+      permissionsAdded: [...desiredState.permissionsAdded],
+      permissionsRemoved,
+    });
+  };
+
+  const handleRestorePermission = (permissionId: string) => {
+    const permissionsRemoved = new Set(desiredState.permissionsRemoved);
+    permissionsRemoved.delete(permissionId);
+    saveToPermissionHistory({
+      permissionsAdded: [...desiredState.permissionsAdded],
+      permissionsRemoved,
+    });
+  };
+
+  const handleSavePermissionsWithConfirm = (confirm: boolean) => {
+    if (confirm) {
+      setConfirming(true);
+      return;
+    }
+
+    const effectiveModifiedPermissions = currentPermissions.filter(
+      (permission) => !desiredState.permissionsRemoved.has(permission.id),
+    );
+
+    const groupPayload = {
+      name: group?.name,
+      description: group?.description,
+      permissions: effectiveModifiedPermissions,
+    };
+
+    setSubmitting(true);
+    updateGroup(groupPayload)
+      .then(() => {
+        void queryClient.invalidateQueries({
+          queryKey: [queryKeys.authGroups],
+        });
+        toastNotify.success(`Permissions for group ${group?.name} updated.`);
+        panelParams.clear();
+        notify.clear();
+      })
+      .catch((e) => {
+        notify.failure("Failed to update permissions", e);
+      })
+      .finally(() => {
+        setSubmitting(false);
+        setConfirming(false);
+      });
+  };
+
+  const closePanel = () => {
+    panelParams.clear();
+    notify.clear();
+  };
+
+  const filteredPermissions = search
+    ? currentPermissions.filter((permission) => {
+        return (
+          permission.entitlement.includes(search) ||
+          permission.entity_type.includes(search) ||
+          permission.url.toLowerCase().includes(search)
+        );
+      })
+    : currentPermissions;
+
+  const imageNamesLookup = generateImageNamesLookup(images);
+  const identityNamesLookup = generateIdentityNamesLookup(identities);
+
+  const sortedPermissions = useMemo(
+    () =>
+      filteredPermissions.sort(
+        generatePermissionSort(imageNamesLookup, identityNamesLookup),
+      ),
+    [filteredPermissions, imageNamesLookup, identityNamesLookup],
+  );
+
+  const modifiedPermissions = getModifiedPermissions();
+
+  const headers = [
+    {
+      content: "Resource type",
+      sortKey: "resourceType",
+      className: "resource-type",
+    },
+    { content: "Resource", sortKey: "resource", className: "resource" },
+    {
+      content: "Entitlement",
+      sortKey: "entitlement",
+      className: "entitlement",
+    },
+    { "aria-label": "Actions", className: "u-align--right actions" },
+  ];
+
+  const rows = sortedPermissions.map((permission, idx) => {
+    const resourceNameFromUrl = extractResourceNameFromUrl(
+      permission.entity_type,
+      permission.url,
+    );
+
+    const resourceName =
+      imageNamesLookup[resourceNameFromUrl] ||
+      identityNamesLookup[resourceNameFromUrl] ||
+      resourceNameFromUrl;
```

Comment:
Maybe provide the imageNamesLookup and identityNamesLookup into the extractResourceNameFromUrl function as additional params, and it can do the lookup for the right types? This is technically incorrect (though I think it would not be very visible) -- because an image and an identiy could have the same name.

Reply:
Sure, I think it should be fine as long as the image id and identity id isn't the same so that we don't result in an incorrect lookup.